awsmp/_driver.py

```diff
@@ -60,21 +60,14 @@ def update_instance_types(
         :rtype: ChangeSetReturnType or None
         """
 
-        changeset, hourly_diff, annual_diff = self._get_instance_type_changeset_and_pricing_diff(offer_config)
+        changeset, hourly_diff, annual_diff = self._get_instance_type_changeset_and_pricing_diff(
+            offer_config, price_change_allowed
+        )
         changeset_name = f"Product {self.product_id} instance type update"
 
         if changeset is None:
             return None
 
-        if hourly_diff or annual_diff:
-            if not price_change_allowed:
-                logger.error(
-                    "There are pricing changes but changing price flag is not set. Please check the pricing files or set the price flag.\n"
-                    "Price change details:\n"
-                    f"Hourly: {hourly_diff}\nAnnual: {annual_diff}\n"
-                )
-                return None
-
```

Comment:
after changes this had zero coverage. as far as i can tell, we can't hit this path anymore due to `_get_pricing_diff` raising in all matching cases